sphinx/builders/linkcheck.py

```diff
@@ -373,6 +385,7 @@ def __init__(
         self.retries: int = config.linkcheck_retries
         self.rate_limit_timeout = config.linkcheck_rate_limit_timeout
         self._allow_unauthorized = config.linkcheck_allow_unauthorized
+        self._timeout_status: Literal['broken', 'timeout']
```

Comment:
This is a nice benefit of the `Literal` approach - narrowing an individual attribute to a subset of the values :+1: